File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        HashMap<String, Integer> m = new HashMap<>();
        
        for(int i = 0; i < participant.length; i++){
            m.put(participant[i], m.getOrDefault(participant[i],0)+1);
        }
        
        for(int i = 0; i < completion.length; i++){
            m.put(completion[i], m.get(completion[i]) - 1);
            if(m.get(completion[i]) == 0)
                m.remove(completion[i]);
        }
        
        Iterator<String> i = m.keySet().iterator();
        
        return answer = i.next();
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):    
    dic = {}
    for p in participant:
        if p not in dic:
            dic[p] = 1
        else:
            dic[p] += 1
    
    for c in completion:
        if dic[c] > 1:
            dic[c] -= 1
        else:
            dic.pop(c)
    
    return list(dic.keys())[0]